web-app/src/pages/Dashboard/container.js

```diff
@@ -33,109 +29,165 @@ const DashboardContainer = Wrapped =>
-            takenHolidays: this.formatDates(response.data),
+            takenHolidays: eventsForCalendar,
           });
         })
         .catch(error => {
           Swal({
-            title: 'Could not get taken holidays',
             text: error.message,
+            title: 'Could not get taken holidays',
             type: 'error',
           });
         });
     };
 
-    formatDates(events) {
-      var eventsForCalendar = events.map(hol => {
-        return {
-          id: hol.holidayId,
-          title: `${hol.employee.forename} ${hol.employee.surname}`,
-          allDay: !hol.halfDay,
-          isHalfday: hol.halfDay,
-          isWFH: false,
-          ...hol,
-          holidayStatusId: hol.holidayStatusId,
-        };
-      });
-      return eventsForCalendar;
-    }
+    getTakenHolidaysById = id => {
+      if (id === 'all') {
+        this.getTakenHolidays();
+      } else {
+        getHolidays(id)
+          .then(response => {
+            const eventsForCalendar = response.data.map(event => {
+              return this.createBookingObj(event);
+            });
+            this.setState(
+              {
+                takenHolidays: eventsForCalendar,
+              },
+              () => this.onFilterEvents(),
+            );
+          })
+          .catch(error => {
+            Swal({
+              text: error.message,
+              title: 'Could not get taken holidays',
+              type: 'error',
+            });
+          });
+      }
+    };
 
     closeModal = () => {
       this.setState({ showModal: false });
     };
 
-    getDuration = booking => {
-      const { isHalfday, isWFH, start, end } = booking;
+    updateBookingAndDuration = booking => {
+      const { isHalfday, eventType, start, end } = booking;
       booking.duration = 1;
       if (isHalfday) {
         booking.duration = 0.5;
-      } else if (isWFH) {
+      } else if (eventType.eventTypeId !== 1) {
         booking.duration = 0;
       } else {
-        booking.duration = Math.floor(
-          moment.duration(end.diff(start)).asDays() + 1,
-        );
+        booking.duration = getDurationNotice(start, end);
       }
 
       this.setState({
-        showModal: true,
         booking: booking,
+        showModal: true,
       });
     };
 
-    onSelectSlot = ({ start, end }) => {
-      let booking = {
-        start: moment(start),
-        end: moment(end),
-        isHalfday: false,
-        isWFH: false,
-        isEventBeingUpdated: false,
-      };
-      this.getDuration(booking);
+    onFilterEmployee = employee => {
+      this.getTakenHolidaysById(employee.employeeId);
     };
 
-    onSelectEvent = booking => {
-      const updatedBooking = {
-        ...booking,
-        isEventBeingUpdated: true,
-      };
-      this.getDuration(updatedBooking);
+    onFilterEvents = eventStatusId => {
+      let updatedFilterEvents = [...this.state.filterEvents];
+      if (eventStatusId !== undefined) {
```

Comment:
Picky, but `if (eventStatusId) {` should be enough here, I think

Reply:
yeah, your right